modules/Dashboard/Http/Controllers/DashboardController.php

```
@@ -541,23 +541,113 @@ private static function _get_perfdata_class($cur, $warn, $crit)
      */
     public function news()
+    {
+        // check for insecure install
+        $sec = $this->news_is_insecure_install();
+        if ($sec)
+            return $sec;
+
+        // Install add sequence check
+        if (\Module::collections()->has('ProvBase'))
+            if (\Modules\ProvBase\Entities\Modem::count() == 0)
+                return $this->news_install_add_sequence_check();
+
+        // Check for official news from nmsprime.com
+        $news = $this->news_load_official_site();
+        if ($news)
+            return $news;
+
+
+        // crowdin - check if language is still supported, otherwise show crowdin link
+        if (!in_array(\Auth::user()->language, config('app.supported_locale')))
+            return ['youtube' => 'https://www.youtube.com/embed/9mydbfHDDP4',
+                    'text' => ' <li>NMS PRIME is not yet translated to your language. Help translating NMS PRIME with
+                    <a href="https://crowdin.com/project/nmsprime/'.\Auth::user()->language.'" target="_blank">Crowdin</a></li>'];
+
+        // links need to be in embedded style, like:
+        //return ['youtube' => 'https://www.youtube.com/embed/9mydbfHDDP4',
+        //		'text' => "You should do: <a href=https://lifeisgood.com>BlaBlaBla</a>"];
+    }
+
+
+    /*
+     * For News Blade:
+     *
+     * Check if installation is secure
+     */
+    private function news_is_insecure_install()
     {
         // change default psw's
         if (\Hash::check('toor', \Auth::user()->password)) {
             return ['youtube' => 'https://www.youtube.com/embed/TVjJ7T8NZKw',
                     'text' => '<li>Next: Change default Password! '.\HTML::linkRoute('User.profile', 'Global Config', \Auth::user()->id), ];
         }
 
+        // check for insecure MySQL root password
+        // This requires to run: mysql_secure_installation
+        if (env('ROOT_DB_PASSWORD') == '') {
```

```php
if (env('ROOT_DB_PASSWORD') != '') {
    return;
}
try {
    if ( \DB::connection('mysql-root')->getPdo()&& \DB::connection()->getDatabaseName()) {
        return ['youtube' => 'https://www.youtube.com/embed/dZWjeL-LmG8',
                    'text' => '<li>Danger! Run: mysql_secure_installation in bash as root!', ];
    } catch (\Exception $e) {
// ??? throw exception or do some output here
}

```

---

if you dont catch the error, there is no reason to use try- catch statement... either you throw an error or do logging or something else. But if nothing is done during an exception, the try-catch could be removed....

---

IMHO: a try without catch does not break the code if a problem happens inside. If code breaks without a try, we will see an error here. This was my goal here

---

AND: i use the try inside the if-statement to let other checks possible run afterwards. Your proposal will break this workflow and skip other tests

---

The if statement is inverted to better indent the code and to clear the code.... because this is essentially what you do. If the root_db_password is not empty, you return from this method.
```php
if (env('ROOT_DB_PASSWORD') != '') {
    return;
}
```
And try with empty catch is an anti-pattern (see [here ](https://stackoverflow.com/questions/4820211/catch-an-exception-without-doing-anything-in-the-catch), [here](https://softwareengineering.stackexchange.com/questions/16807/is-it-ever-ok-to-have-an-empty-catch-statement/16822) or [here](https://stackoverflow.com/questions/17838143/try-with-empty-catches)), you should at least log the exception. Because if it fails, its is for a reason
I dont know, if this is 100% correct, but would make more sense to me:
```php
try {
    \DB::connection('mysql-root')->getPdo();
    \DB::connection()->getDatabaseName()
    } catch (Exception $e) {
        return [
            'youtube' => 'https://www.youtube.com/embed/dZWjeL-LmG8',
            'text' => '<li>$e->message - Run: mysql_secure_installation in bash as root!', 
        ];
}
```